src/table/src/predicate.rs

```diff
@@ -334,6 +380,165 @@ mod tests {
 
     use super::*;
 
+    fn check_build_predicate(expr: DfExpr, expect: TimestampRange) {
+        assert_eq!(
+            expect,
+            TimeRangePredicateBuilder::new("ts", TimeUnit::Millisecond, &[Expr::from(expr)])
+                .build()
+        );
+    }
+
+    #[test]
+    fn test_build_predicate() {
+        // operator GT
+        // ts > 1ms
+        check_build_predicate(
+            col("ts").gt(lit(ScalarValue::TimestampMillisecond(Some(1), None))),
+            TimestampRange::from_start(Timestamp::new_millisecond(2)),
+        );
+
+        // 1ms > ts
+        check_build_predicate(
+            lit(ScalarValue::TimestampMillisecond(Some(1), None)).gt(col("ts")),
+            TimestampRange::until_end(Timestamp::new_millisecond(1), false),
+        );
+
+        // 1001us > ts
+        check_build_predicate(
+            lit(ScalarValue::TimestampMicrosecond(Some(1001), None)).gt(col("ts")),
+            TimestampRange::until_end(Timestamp::new_millisecond(1), true),
+        );
+
+        // ts > 1001us
+        check_build_predicate(
+            col("ts").gt(lit(ScalarValue::TimestampMicrosecond(Some(1001), None))),
+            TimestampRange::from_start(Timestamp::new_millisecond(2)),
+        );
+
+        // 1s > ts
+        check_build_predicate(
+            lit(ScalarValue::TimestampSecond(Some(1), None)).gt(col("ts")),
+            TimestampRange::until_end(Timestamp::new_millisecond(1000), false),
+        );
+
+        // ts > 1s
+        check_build_predicate(
+            col("ts").gt(lit(ScalarValue::TimestampSecond(Some(1), None))),
+            TimestampRange::from_start(Timestamp::new_millisecond(1001)),
+        );
+
+        // operator GT_EQ
+        // ts >= 1ms
```

Comment:
nitpick(non-blocking): Could we document the expected range in the comment?
```suggestion
        // ts >= 1ms -> [1ms, ..)
```

Reply:
> nitpick(non-blocking): Could we document the expected range in the comment?

I have to distinguish `ts >= [some_literal]` and `[some_literal] >= ts`